tests/test_global.py

```diff
@@ -66,12 +162,52 @@ def test_predict_all_function():
 
                     falses = falses + 1
 
-            print(
-                "False rating off: "
-                + str(falses / len(glob.glob(output_path_test + "*SV*.parquet")))
-            )
+                print(
+                    "False rating off: "
+                    + str(falses / len(glob.glob(output_path_test + "*PNEO*.parquet")))
+                )
 
-            os.remove(
-                output_path + a_tif_file.split("/")[-1].replace(".tif", ".parquet")
-            )
-            assert falses == 0
+                os.remove(afile)
+
+    assert falses == 0
+
+
+def test_hexagon():
+
+    with contextlib.redirect_stdout(io.StringIO()):
+        output_file_name_generator = OutputFileNameGenerator(
+            output_path=output_path_test,
+            output_file_name=output_path_test
+            + a_tif_file_hexagon_test.split("/")[-1].replace(".tif", ".parquet"),
+        )
+
+    final_artefact = pickle.load(open(model_path_pneo, "rb"))
+    selected_features = ["r", "g", "b", "n", "e", "d", "ndvi", "re_ndvi"]
+
+    nso_tif_kernel_iterator_generator = tif_model_iterator.TifModelIteratorGenerator(
+        path_to_tif_file=a_tif_file_hexagon_test,
+        model=final_artefact["model"],
+        output_file_name_generator=output_file_name_generator,
+        parts=1,
+        normalize_scaler=final_artefact["scaler"],
+        column_names=selected_features,
+        dissolve_parts=False,
+        square_output=False,
+        skip_done_part=False,
+    )
+
+    nso_tif_kernel_iterator_generator.predict_all_output()
+    print(
+        "File stored here: "
+        + str(nso_tif_kernel_iterator_generator.get_final_output_path())
+    )
+
+    path_hexagons = output_h3_hexagons_from_pixels(
+        nso_tif_kernel_iterator_generator.get_final_output_path(),
+        13,
+    )
+
+    file_exists = is_valid_geodataframe_with_polygons(gpd.read_file(path_hexagons))
+    os.remove(path_hexagons)
```

Comment:
Maybe a bit too much now, but do you know about setup and teardown in testing?

Examples for pytest can be found here:
https://stackoverflow.com/questions/26405380/how-do-i-correctly-setup-and-teardown-for-my-pytest-class-with-tests